includes/data-stores/class-wc-admin-reports-orders-data-store.php

```diff
@@ -470,4 +486,28 @@ protected static function get_num_items_sold( $order ) {
 
 		return $num_items;
 	}
+
+	/**
+	 * Check to see if an order's customer has made previous orders or not
+	 *
+	 * @param array $order WC_Order object.
+	 * @return bool
+	 */
+	protected static function is_returning_customer( $order ) {
+		$customer_id = $order->get_user_id();
+
+		if ( 0 === $customer_id ) {
+			return false;
+		}
+
+		$customer_orders = get_posts( array(
+			'meta_key' => '_customer_user',
+			'meta_value' => $customer_id,
+			'post_type' => 'shop_order',
+			'post_status' => array( 'wc-on-hold', 'wc-processing', 'wc-completed' ),
+			'numberposts' => 2,
+		) );
+
+		return count( $customer_orders ) > 1;
+	}
```

Comment:
I initially thought about using a similar approach, but then was told about [possible problems](https://protonpower.wordpress.com/2018/06/13/rest-api-v3-progress-mocking-episode-2/#comment-2348) (TL;DR: this would only work for registered customers, we could probably base this on billing email, but that can't be done on the fly for performance reasons).

Also, I think this would mark first order for each customer as order from a new customer, even if the customer ordered again. Once the customer orders for the second time, also the first order becomes an order from a returning customer.

No definitive decision has been made, but maybe we can try to implement the approach as outlined in the comment on p2 and use the wc_order_stats to store the order flag?
I.e.:
```
When making an order, we could check (maybe in the bg queue if performance costly) if this customer already has any fulfilled order and then

if the count of orders > 1, then we just flag this one as from returning customer,
if the count of orders = 1, we flag this one as from returning, plus update the first order from the customer to returning
if the count of orders = 0, we flag as new customer
```

Reply:
Good points, Peter.  Here are my thoughts on each:

#### Guests

Counting guests is an interesting dilemma.  

If we used the solution in this PR, I don't think we have the same performance concerns since this is only synced on `save_post` but my current implementation isn't really correct since it really only counts new/returning orders and not new/returning customers.

Joining on `wp_wc_order_product_lookup` for `customer_id` would resolve this for non-guests, but for guests this only allows us to check new/returning orders and not customers.

#### Updating returning orders

Maybe my thought process on this is a bit off, but wouldn't we want to keep the initial order as a new customer?  

If I'm a store owner I think that if a new customer placed 2 orders within the period I'm searching, I'd still want to see them as a new customer if their first order falls within the timeframe I've selected since I would use this information to guage marketing efforts.

What are your thoughts on this?